jenkins/docker/ubuntu/docs.Dockerfile

```diff
@@ -0,0 +1,71 @@
+ARG FROM_IMAGE='ubuntu:24.04'
+
+FROM ${FROM_IMAGE}
+
+# install dependencies (install tzdata first to prevent 'geographic area' prompt)
+RUN apt-get update \
+	&& apt-get install -y tzdata \
+	&& apt-get install -y openjdk-11-jdk-headless git curl libssl-dev maven ca-certificates zip unzip \
+	&& update-ca-certificates
+
+# install python
+RUN apt-get install -y python3-pip python3-venv
+
+# install shellcheck
+RUN apt-get install -y shellcheck
+
+# nodejs
+ENV NODE_OPTIONS="--dns-result-order=ipv4first"
+ARG NODEJS_VERSION=22
+RUN apt-get install -y ca-certificates curl gnupg \
+	&& mkdir -p /etc/apt/keyrings \
+	&& curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg \
+	&& echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_${NODEJS_VERSION}.x nodistro main" \
+	| tee /etc/apt/sources.list.d/nodesource.list \
+	&& apt-get update \
+	&& apt-get install -y nodejs
+
+RUN npm install -g typedoc typedoc-plugin-markdown
+
+# doxygen
+RUN apt-get install -y bison cmake flex gcc g++ graphviz make qtcreator qt6-tools-dev texlive-full
+ARG DOXYGEN_VERSION=1.13.2
+RUN curl -Os https://www.doxygen.nl/files/doxygen-${DOXYGEN_VERSION}.src.tar.gz \
+	&& tar -xf doxygen-${DOXYGEN_VERSION}.src.tar.gz \
+	&& cd doxygen-${DOXYGEN_VERSION} \
+	&& mkdir build \
+	&& cd build \
+	&& cmake -G "Unix Makefiles" .. \
+	&& cmake -Dbuild_wizard=YES .. \
+	&& make \
+	&& make install \
+	&& cd ../.. \
+	&& rm -rf doxygen-${DOXYGEN_VERSION} doxygen-${DOXYGEN_VERSION}.src.tar.gz
+
+# codecov uploader
```

Comment:
do you need codecov?

Reply:
Yeah, it probably can be removed, but I wasn't sure.